slack_today_i_did/our_repo.py

```diff
@@ -99,21 +102,82 @@ def get_files_for_017(self, pattern: str) -> List[str]:
             if self.what_kinda_file(filename) == ElmVersion.v_017
         ]
 
-    def get_017_porting_breakdown(self, pattern: str) -> Dict[str, Dict[str, int]]:  # noqa: E501
+    def get_matching_filenames(self, pattern: str) -> List[str]:
         pattern = pattern.replace('.', '/')
         all_files = glob.glob(
             f'{self.repo_dir}/**/{pattern}.elm',
             recursive=True
         )
+        return all_files
+
+    @contextmanager
+    def lock(self) -> None:
+        self.create_lock()
+        yield
+        self.remove_lock()
+
+    def create_lock(self) -> None:
+        """ create a lock so that caches can be used to make some calcs faster """
+        self._breakdown_lock = True
+        self._breakdown_cache = {}
```

Comment:
💭 It's a little unintuitive to me that locking a repo would clear the cache.

When I see "lock", I imagine it'd do some sort of access prevention. Here, the intention seems to be more about encouraging side-effects.

How about:

```python
    @contextmanager
    def cache_lookups(self) -> None:
        self.start_caching()
        yield
        self.stop_caching()

    def start_caching(self) -> None:
        """ create caches to make some calcs faster """
        self._caching_lookups = True
        self._breakdown_cache = {}
        self._import_cache = {}
```